internal/healthchecks/healthCheck.go

```diff
@@ -17,19 +22,35 @@ func checkServerHealth(server string, healthCheckConfig config.HealthCheck) bool
 	return err == nil && res.StatusCode == 200
 }
 
-func PerformHealthChecks(config *config.Config) {
+func worker(id int, tasks <-chan HealthCheckTask) {
+	for task := range tasks {
+		if !checkServerHealth(task.Server, task.HealthCheckConfig) {
+			log.Printf("[Worker %d] Server %s failed health check, removing from pool\n", id, task.Server)
+		} else {
+			log.Printf("[Worker %d] Server %s passed health check\n", id, task.Server)
+		}
```

Comment:
it's not enough to log this information right?

You need to update the health status somewhere

Reply:
Used a map to store the health stasus of each server